test/gtest/smooth_l1loss.hpp

```diff
@@ -0,0 +1,197 @@
+    {
+        auto N    = prng::gen_A_to_B(1, 50);
+        auto C    = prng::gen_A_to_B(0, 100);
+        auto D    = prng::gen_A_to_B(0, 100);
+        auto H    = prng::gen_A_to_B(0, 100);
+        auto W    = prng::gen_A_to_B(0, 100);
+        auto beta = prng::gen_A_to_B(0.0f, 4.0f);
+        tcs.push_back({N, C, D, H, W, MIOPEN_LOSS_NO_REDUCTION, beta});
+        // tcs.push_back({N, C, D, H, W, MIOPEN_LOSS_MEAN_REDUCTION, beta});
+        // tcs.push_back({N, C, D, H, W, MIOPEN_LOSS_SUM_REDUCTION, beta});
+    }
+    return tcs;
+}
+
+template <typename T = float>
+struct SmoothL1LossTest : public ::testing::TestWithParam<SmoothL1LossTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle        = get_handle();
+        smooth_l1loss_config = GetParam();
+        auto gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+
+        reduction = smooth_l1loss_config.reduction;
+        beta      = smooth_l1loss_config.beta;
+
+        auto dims = smooth_l1loss_config.GetInput();
+
+        input  = tensor<T>{dims}.generate(gen_value);
+        target = tensor<T>{dims}.generate(gen_value);
+
+        output = tensor<T>{dims};
+        std::fill(output.begin(), output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        ref_output = tensor<T>{dims};
+        std::fill(ref_output.begin(), ref_output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        std::vector<size_t> workspace_dims;
+        ws_sizeInBytes = miopen::GetSmoothL1LossWorkspaceSize(
+            handle, reduction, input.desc, target.desc, output.desc);
+        if(ws_sizeInBytes > 0)
+        {
+            workspace = tensor<T>{dims};
+            std::fill(workspace.begin(), workspace.end(), std::numeric_limits<T>::quiet_NaN());
+            workspace_dev = handle.Write(workspace.data);
+        }
+
+        input_dev  = handle.Write(input.data);
+        target_dev = handle.Write(target.data);
+        output_dev = handle.Write(output.data);
+    }
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        cpu_smooth_l1loss_forward<T>(input, target, ref_output, reduction, beta);
+        miopenStatus_t status;
+
+        status = miopen::SmoothL1LossForward(handle,
+                                             reduction,
+                                             input.desc,
+                                             input_dev.get(),
+                                             target.desc,
+                                             target_dev.get(),
+                                             output.desc,
+                                             output_dev.get(),
+                                             beta);
+
+        EXPECT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
```

Comment:
I fixed this function into this:
```c++
void Verify()
{
    // Computation error of fp16 is ~2^13 (=8192) bigger than
    // the one of fp32 because mantissa is shorter by 13 bits.
    double tolerance = std::is_same<T, float>::value ? 1.5e-6 : 8.2e-3;

    // bf16 mantissa has 7 bits, by 3 bits shorter than fp16.
    if(std::is_same<T, bfloat16>::value)
        tolerance *= 8.0;

    auto error = miopen::rms_range(ref_output, output);

    EXPECT_TRUE(miopen::range_distance(ref_output) == miopen::range_distance(output));
    EXPECT_TRUE(error < tolerance)
        << "Error output beyond tolerance Error:" << error << ",  Tolerance: " << tolerance;
}
```